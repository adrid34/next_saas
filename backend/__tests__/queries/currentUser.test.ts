import { setup, teardown, runGraphQLQuery, deleteTestUsers } from "../helpers";

beforeEach(deleteTestUsers);
beforeAll(setup);
afterAll(teardown);

test("currentUser", async () => {
  await runGraphQLQuery(
    // GraphQL query goes here:
    `{currentUser{id}}`,

    // GraphQL variables:
    {},

    // Additional props to add to `req` (e.g. `user: {session_id: '...'}`)
    {
      user: null,
    },

    // This function runs all your test assertions:
    async json => {
      expect(json.errors).toBeFalsy();
      expect(json.data).toBeTruthy();
      expect(json.data!.currentUser).toBe(null);
    }
  );
});
