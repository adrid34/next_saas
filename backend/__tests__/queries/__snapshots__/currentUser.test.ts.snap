// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currentUser 1`] = `
Object {
  "data": Object {
    "currentUser": null,
  },
}
`;
